api-remove.go

```diff
@@ -29,6 +29,52 @@ import (
 	"github.com/minio/minio-go/v7/pkg/s3utils"
 )
 
+// BucketOptions special headers to purge buckets, only
+// useful when endpoint is MinIO
+type BucketOptions struct {
+	ForceDelete bool
+}
+
+// RemoveBucketWithOptions deletes the bucket name, additionally
+// takes options to force purging a bucket or a prefix when the
+// endpoint is MinIO server
+//
+// All objects (including all object versions and delete markers)
+// in the bucket will be deleted if bucket options set ForceDelete
+// as true.
+func (c Client) RemoveBucketWithOptions(ctx context.Context, bucketName string, opts BucketOptions) error {
+	// Input validation.
+	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
```

Comment:
The comment says it can remove a prefix but we are validating this as a bucket name, maybe what we can do is we call this function: RemoveWithOptions() and add a Prefix string in BucketOptions, which can be called RemoveOptions I guess.

Reply:
BucketName is the most important requirement - prefix doesn't need validation since its allowed to be empty.

Reply:
That comment is wrong